File: widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadLayoutContract.java
package com.glia.widgets.view.head;

import com.glia.widgets.base.BaseController;
import com.glia.widgets.base.BaseView;

public interface ChatHeadLayoutContract {
    interface Controller extends BaseController {
        void onChatHeadClicked();

        void setView(View view);
    }

    interface View extends BaseView<ChatHeadLayoutContract.Controller> {
        void showOperatorImage(String operatorImgUrl);

        void showUnreadMessageCount(int count);

        void showPlaceholder();

        void showQueueing();
        void showScreenSharing();

        void showOnHold();

        void hideOnHold();

        void navigateToChat();

        void navigateToCall();

        void navigateToEndScreenSharing();

        boolean isInChatView();

        void show();

        void hide();
    }
}

File: widgetssdk/src/main/java/com/glia/widgets/view/head/ChatHeadLayoutContract.kt
package com.glia.widgets.view.head

import com.glia.widgets.base.BaseController
import com.glia.widgets.base.BaseView

internal interface ChatHeadLayoutContract {
    interface Controller : BaseController {
        fun onChatHeadClicked()
        fun onResume(viewName: String)
        fun setView(view: View)
        fun shouldShow(gliaOrRootViewName: String?): Boolean
        fun updateChatHeadView()
        fun onPause(viewName: String)
    }

    interface View : BaseView<Controller> {
        fun showOperatorImage(operatorImgUrl: String)
        fun showUnreadMessageCount(count: Int)
        fun showPlaceholder()
        fun showQueueing()
        fun showOnHold()
        fun hideOnHold()
        fun navigateToChat()
        fun navigateToCall()
        fun show()
        fun hide()
    }
}
